### ISSUE TITLE
Transparency can't be lightened or darkened and no error message

### ISSUE THREAD

**Issue by lukeaus** [less/less.js#2588]
Now I am aware that its illogical to lighten or darken transparency...

that being said, it would be good if an attempt to do so threw an error

Im using an older version of lessc (2.0.0-beta.0)so this may have been fixed already, but I couldn't find any existing issues.

I had a function:
```less
.btn-common (
    @color-text: @white,
    @color-bg: @brand-primary,
    @color-border-top: lighten(@color-bg, 5%),
    @color-border-bottom: darken(@color-bg, 20%),
  ) {
  .BorderRadius;
  .NoBorderNoShadow;
  color: @color-text !important;
  background-color: @color-bg !important;
  background-image: none !important;
  border-left: none;
  border-right: none;
  border-top: 2px solid @color-border-top !important;
  border-bottom: 2px solid @color-border-bottom !important;
}
```
and then I was calling this function
```less
  .btn-common(@drkgrey, transparent);
```

which didn't work, and took me a while to figure out what was going on

this does work however
```less
  .btn-common(@drkgrey,
              transparent,
              transparent,
              transparent);
```
So basically it would be good to throw an error if attempting to lighten or darken transparent

Thanks team Less!

**Comment by seven-phases-max** [comment]
Please use [backticks](https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks) to format you code. Also it is always important what you mean with "not working" - e.g. you you expect to get and what you get instead.
I test you example and it compiles to a generally fine result:
```less
lighten(transparent, 5%) // -> rgba(13, 13, 13, 0)
darken(transparent, 20%) // -> rgba(0, 0, 0, 0)
```
e.g. obviously a lighter or darker transparent is still a transparent as expected. Is your problem is specific to some older browser, which does support `transparent` keyword but does not support `rgba`? (in that sense it would make sense to always render any fully transparent colour transparent as `transparent` instead of `rgba(*, *, *, 0).. Though I wonder what browser it is... ?)

Aside of above the code (both Less and CSS)  is absolutely valid so it should not throw any error.

**Comment by lukeaus** [comment]
Hi seven-phases-max,
The error was that a compiled .css file was not being generated.

On top of that, no error was being raised.

Interestingly the code would work locally but would not work on our production server.

Tested working locally with FF, Chrome and Safari v7.1

Failed on the production server with Chrome and Safari v7.1 (don't think I tested with Firefox),

We are using django and django-compress v1.3

Bit of a mystery to me atm...any ideas?

**Comment by seven-phases-max** [comment]
>We are using django and django-compress v1.3

Both local and at production server? If so I'd started to look what are all those differences in between (configurations, versions and so on).

Speaking of 
>On top of that, no error was being raised.

Less itself never bails out w/o generating a error so if it's suppressed it's something on the `django` side.
So I'm marking thi for closing as not a Less issue (unless you can reproduce the same behaviour with  `lessc/less.js` or so).
